test/functional/device/calendar-e2e-specs.js

```diff
@@ -1,40 +1,43 @@
 import chai from 'chai';
 import chaiAsPromised from 'chai-as-promised';
-import { TESTAPP_CAPS } from '../desired';
+import { TESTAPP_CAPS, amendCapabilities } from '../desired';
 import { initSession, deleteSession, MOCHA_TIMEOUT } from '../helpers/session';
 
 
 chai.should();
 chai.use(chaiAsPromised);
 
 if (!process.env.REAL_DEVICE && !process.env.CLOUD) {
-  describe('XCUITestDriver - calendar', function () {
+  describe('XCUITestDriver - calendar', function() {
     this.timeout(MOCHA_TIMEOUT);
 
     let driver, caps;
 
-    beforeEach(function () {
+    beforeEach(function() {
       caps = Object.assign({}, TESTAPP_CAPS);
     });
 
-    afterEach(async function () {
+    afterEach(async function() {
+      await driver.terminateApp('io.appium.TestApp');
       await deleteSession();
     });
 
-    it('should authorize calendar access if calendarAccessAuthorized == true', async function () {
-      caps.calendarAccessAuthorized = true;
+    it('should authorize calendar access if calendarAccessAuthorized == true', async function() {
+      caps = amendCapabilities(caps, { 'appium:calendarAccessAuthorized': true });
       driver = await initSession(caps);
-      let checkCalendarButton = await driver.elementByName('Check calendar authorized');
+      let checkCalendarButton = await driver.$("-ios predicate string:name='Check calendar authorized'");
```

Comment:
Consider using backticks instead of double quotes if the string already has single quotes

Reply:
also, search by name equals to accessibility id in ios